functions.sh

```diff
@@ -85,28 +85,42 @@ copy_file() {
     run_with_timeout 1000s "virt-copy-out ${disks} ${file} ${dir}"
 }
 
-validate_RESULT() {
-    disksdir=$1
-    args=$(for d in ${disksdir}/disk-*img; do echo -a ${d}; done)
-
-    # Use also iscsi disk if there is any
-    if [[ -n ${iscsi_disk_img} ]]; then
-        args="${args} -a ${disksdir}/${iscsi_disk_img}"
-    fi
+copy_interesting_files_from_system() {
+    args="$1"
+    disksdir="$2"
 
     # Grab files out of the installed system while it still exists.
     # Grab these files:
     #
     # logs from Anaconda - whole /var/log/anaconda/ directory is copied out,
     #                      this can be used for saving specific test output
-    # anaconda.coverage
-    # RESULT file from the test
-    for item in /root/anaconda.coverage \
-                /var/log/anaconda/      \
+    # original-ks.cfg - the kickstart used for the test
+    # anaconda-ks.cfg - the kickstart saved after installation, useful for
+    #                   debugging
+    # RESULT - file from the test
+    for item in /root/original-ks.cfg \
+                /root/anaconda-ks.cfg \
+                /var/log/anaconda/    \
                 /root/RESULT
     do
-        copy_file "${args}" "${item}" "${disksdir}"
+        if [[ -z "$3" ]]; then
+            copy_file "${args}" "${item}" "${disksdir}"
+        else
+            copy_file "${args}" "${item}" "${disksdir}" 2>/dev/null
```

Comment:
Maybe we could always redirect stderr to /dev/null to simplify the code a little.

Reply:
It would be definitely nicer, but I am not sure if it's a good idea. The core problem is: This is normally called by the tests, and then you want to see the errors. But it is also called by the "nosave" tests, and I don't understand why these are different.